cloudinit/net/networkd.py

```diff
@@ -46,33 +48,57 @@ def update_route_section(self, sec, rid, key, val):
             if k == sec:
                 if rid not in self.conf_dict[k]:
                     self.conf_dict[k][rid] = []
-                self.conf_dict[k][rid].append(key + "=" + str(val))
+                self.conf_dict[k][rid].append(f"{key}={val}")
                 # remove duplicates from list
                 self.conf_dict[k][rid] = list(
                     dict.fromkeys(self.conf_dict[k][rid])
                 )
                 self.conf_dict[k][rid].sort()
 
+    def normalize_dict(self, data):
```

Comment:
The recursive nature of this introduces some typing issues. `data` can be a dict or list, which means `normalize_dict()` can return a dict or list. I think it might be best to introduce a helper function while having the top level function take `Dict[str, List]` and return `Dict[str, List]`.

Also, the original declaration of `self.conf_dict` no longer needs to be an OrderedDict now, correct?